### DISCUSSION TITLE
[Question] DGML Support

### DISCUSSION THREAD

**Discussion by shaggygi** [VsixCommunity/Community.VisualStudio.Toolkit#73] [Q&A]
Any thought around including DGML, how to simplify, and different use-cases?

For instance (and with limited knowledge if there are other alternatives), would it be appropriate to use DGML to create different UI like a TreeView of custom elements (similar to Solution Explorer)?

Came across this repo that attempts to simplify.
https://github.com/merijndejonge/DgmlBuilder

### References
https://docs.microsoft.com/en-us/visualstudio/modeling/directed-graph-markup-language-dgml-reference?view=vs-2019

**Comment by madskristensen** [comment]
There is a WPF tree view control. Could that not be used?

**Comment by shaggygi** [comment]
> There is a WPF tree view control. Could that not be used?

Updated:
Yes, but I was thinking there might be more specific controls/APIs within the SDK to help build up the items.  For example, are there APIs/examples to convert a DirectedGraph to a TreeView?

And even if the WPF control helps with some stuff I'm looking at using for, seems like the toolkit could still possibly include helpers for DGML.  Wasn't aware of DGML until this morning and seems very powerful/helpful on some things I need.

**Comment by StevenRasmussen** [comment]
Yeah... I'm not sure this would be a good fit for this project.  I haven't worked with DGML files though... I guess if there were some specific API's relating to DGML in the SDK that you feel could be simplified/improved upon then that would probably be something to consider.  Do you have any specific examples of API's or code that you currently use for working with DGML files that you feel could benefit by being wrapped into helper functions and included here?

**Comment by shaggygi** [comment]
> Do you have any specific examples of API's or code that you currently use for working with DGML files that you feel could benefit by being wrapped into helper functions and included here?

Not specifically yet.  Still trying to learn/understand them.  Didn't know they existed until a couple days ago, but seem very helpful (and complex).  This might be a topic where the core SDK improves the docs explaining all related pieces.  I've found a few related blogs and @madskristensen provided a link to one of his extensions that use, so starting there for now.
